code/modules/shuttle/shuttle.dm

```diff
@@ -704,6 +750,8 @@
 			if(prearrivalTime && mode != SHUTTLE_PREARRIVAL)
 				set_mode(SHUTTLE_PREARRIVAL)
 				setTimer(prearrivalTime)
+				log_debug("SHUTTLE DEBUG: shuttle on approach failed check!")
+				on_prearrival()
 				return
 			on_prearrival()
```

Comment:
Document modifications from upstream code clearly please, this avoids things getting reverted on update